src/App.tsx

```diff
@@ -1,61 +1,130 @@
+import { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const initialTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+const maxTodoId = initialTodos
+  .map(todo => todo.id)
+  .sort((a, b) => a - b)[initialTodos.length - 1];
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [todos, setTodos] = useState(initialTodos);
+  const [todoId, setTodoId] = useState(maxTodoId);
+  const [isEmptyTitle, setIsEmptyTitle] = useState(false);
+  const [isEmptyUser, setIsEmptyUser] = useState(false);
+
+  const changeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    if (event.target.value.trim() !== '') {
+      setIsEmptyTitle(false);
+    }
+  };
+
+  const changeId = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    if (+event.target.value !== 0) {
+      setIsEmptyUser(false);
+    }
+  };
+
+  const handleSubmit = (event: React.ChangeEvent<HTMLFormElement>) => {
+    event?.preventDefault();
+    if (title.trim() === '' || userId === 0) {
+      if (title.trim() === '') {
+        setIsEmptyTitle(true);
+      }
+
+      if (userId === 0) {
+        setIsEmptyUser(true);
+      }
+
+      return;
+    }
+
```

Comment:
For what did you wrap if in if?
why you cant use only 2 'if'?
like:
      if (title.trim() === '') {
        setIsEmptyTitle(true);
      }

      if (userId === 0) {
        setIsEmptyUser(true);
      }

Reply:
here is the reason on the screenshot, it is needed in order to show the error for both cases if the user is not chosen and the title is not entered and stop the function by return
![Screenshot_101](https://user-images.githubusercontent.com/63102835/202506268-16e3aeba-b908-422a-9a0a-ff19a5d3be58.jpg)